public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -153,13 +183,83 @@ function addPromGroupsToCombinedNamespace(namespace: CombinedRuleNamespace, grou
       const existingRule = getExistingRuleInGroup(rule, combinedRulesByName, namespace.rulesSource);
       if (existingRule) {
         existingRule.promRule = rule;
+        existingRule.instanceTotals = isAlertingRule(rule) ? calculateRuleTotals(rule) : {};
+        existingRule.filteredInstanceTotals = isAlertingRule(rule) ? calculateRuleFilteredTotals(rule) : {};
       } else {
         combinedGroup!.rules.push(promRuleToCombinedRule(rule, namespace, combinedGroup!));
       }
     });
   });
 }
 
+export function calculateRuleTotals(rule: Pick<AlertingRule, 'alerts' | 'totals'>): AlertInstanceTotals {
+  const result = countBy(rule.alerts, 'state');
+
+  if (rule.totals) {
+    const { normal, ...totals } = rule.totals;
+    return { ...totals, inactive: normal };
+  }
+
+  return {
+    alerting: result[AlertInstanceTotalState.Alerting],
+    pending: result[AlertInstanceTotalState.Pending],
+    inactive: result[AlertInstanceTotalState.Normal],
+    nodata: result[AlertInstanceTotalState.NoData],
+    error: result[AlertInstanceTotalState.Error] + result['err'], // Prometheus uses "err" instead of "error"
```

Comment:
I feel like this is prone to error if either of these is `undefined`, let's try to guard against that with `(result[AlertInstanceTotalState.Error] + result['err']) || 0`

Reply:
Should it instead be `result[AlertInstanceTotalState.Error] ?? result['err']`? I think we don't need to sum there as we receive `Error` or `'err` but not both